build.gradle

```diff
@@ -0,0 +1,168 @@
+        testCompile "junit:junit:${junitVersion}"
+        testCompile "org.mockito:mockito-core:${mockitoVersion}"
+    }
+
+    compileScala {
+        scalaCompileOptions.additionalParameters = [
+                "-feature",
+                "-target:jvm-1.8",
+                "-deprecation",
+                "-language:implicitConversions",
+                "-unchecked",
+                // TODO: (fdaca 2019-11-18) please consider repairing warnings and enabling this
+                //"-Xfatal-warnings"
+        ]
+    }
+
+    test {
+        maxHeapSize = '2g'
+        testLogging {
+            events 'started', 'failed', 'passed', 'skipped'
+            showStandardStreams = true
+        }
+    }
+
+    tasks.withType(JavaCompile) {
+        options.encoding = 'UTF-8'
+    }
+}
+
+subprojects {
+    ext.gitCommitIdAbbrev = System.getenv('GIT_COMMIT_ID_ABBREV') ?: 'git rev-parse --verify --short=12 HEAD'.execute().text.trim()
+
+    task sourcesJar(type: Jar, dependsOn: classes) {
+        archiveClassifier.set('sources')
+        from(sourceSets.main.allJava) {
+            expand(project: [version: project.version], git: [commit: [id: [abbrev: gitCommitIdAbbrev]]])
+        }
+        from(sourceSets.main.allScala)
+    }
+
+    task javadocJar(type: Jar, dependsOn: scaladoc) {
+        from javadoc
+        from scaladoc
+        archiveClassifier.set('javadoc')
+    }
+
+    license {
+        header rootProject.file('src/etc/header.txt')
+        strictCheck true
+        exclude "**/warning.txt"
+    }
+
+    publishing {
+        publications {
+            maven(MavenPublication) {
+                groupId(project.group)
+                artifactId(project.name + "_${scalaMajorVersion}")
+                version(project.version)
+
+                pom {
+                    name = project.name
+                    description = project.description
+                    url = 'https://github.com/SumoLogic/shellbase'
+
+                    licenses {
+                        license {
+                            name = 'Apache License, Version 2.0'
+                            url = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+                        }
+                    }
+                    scm {
+                        connection = 'scm:git:git@github.com:SumoLogic/shellbase.git'
+                        developerConnection = 'scm:git:git@github.com:SumoLogic/shellbase.git'
+                        url = 'https://github.com/SumoLogic/shellbase'
+                        tag = 'HEAD'
+                    }
+                }
+
+                from(components.java)
+                artifact(sourcesJar)
+                artifact(javadocJar)
+            }
+        }
+        repositories {
+            maven {
+                url 'https://oss.sonatype.org/service/local/staging/deploy/maven2/'
+                credentials {
+                    username project.findProperty('nexus_username')
```

Comment:
I suggest null-safety here, as I saw it might fail in some contexts (Travis and local build) if the properties are not set. See https://github.com/SumoLogic/sumobot/blob/master/build.gradle#L153

Reply:
I think I made it optional in this commit (tested locally and on Travis): https://github.com/SumoLogic/shellbase/pull/42/commits/5b70021fc4ce5076d7df0821db0bc9d04eb26326

Before that, Travis and local was failing when the property was not set. Now the credentials would be set to `null` without throwing an exception.

Reply:
OK